From bdae4110b6c643e56c10067b9d8cc811713eb01a Mon Sep 17 00:00:00 2001
From: Jay Tu <jay.tu@freescale.com>
Date: Fri, 14 Dec 2012 11:43:58 +0800
Subject: [PATCH 4/4] ENGR00237044 Adds baud rate B3500000

For stty to set larger baud rate since uart can support up to 4M.

Signed-off-by: Jay Tu <jay.tu@freescale.com>
---
 libbb/speed_table.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libbb/speed_table.c b/libbb/speed_table.c
index 05fe66c..6fc3d04 100644
--- a/libbb/speed_table.c
+++ b/libbb/speed_table.c
@@ -3,6 +3,7 @@
  * compact speed_t <-> speed functions for busybox
  *
  * Copyright (C) 2003  Manuel Novoa III  <mjn3@codepoet.org>
+ * Copyright (C) 2012 Freescale Semiconductor, Inc.
  *
  * Licensed under GPLv2 or later, see file LICENSE in this tarball for details.
  */
@@ -11,7 +12,7 @@
 
 struct speed_map {
 	unsigned short speed;
-	unsigned short value;
+	unsigned int value;
 };
 
 static const struct speed_map speeds[] = {
@@ -51,6 +52,7 @@ static const struct speed_map speeds[] = {
 #ifdef B460800
 	{B460800, 460800/256 + 0x8000U},
 #endif
+	{B3500000, 3500000},
 };
 
 enum { NUM_SPEEDS = ARRAY_SIZE(speeds) };
-- 
1.8.0

